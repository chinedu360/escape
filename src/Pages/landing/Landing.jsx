// Import necessary modules and components
import { Button } from "@chakra-ui/react"; // Import a button component from Chakra UI
import { BsArrowRightShort } from "react-icons/bs"; // Import an arrow icon from react-icons
import { useNavigate } from "react-router-dom"; // Import the useNavigate hook from react-router-dom

// Define the Landing component
const Landing = () => {
  const navigate = useNavigate(); // Initialize the navigate function from react-router-dom

  const handleDashBoard = () => {
    navigate("/dashboard"); // Navigate to the "/dashboard" route when the button is clicked
  };

  return (
    <div className="flex flex-col justify-center items-center">
      <div className="">
        <h1 className="font-bold text-[30px] sm:text-[60px] md:text-[80px]">
          Read An Article Today!
        </h1>
      </div>
      <Button
        rightIcon={<BsArrowRightShort />} // Display the arrow icon on the button
        colorScheme="teal" // Set the button color scheme to "teal"
        variant="outline" // Use the outlined button variant
        width="8rem" // Set the button width
        marginTop="2rem" // Add top margin to the button
        size="md" // Set the button size to medium
        onClick={handleDashBoard} // Call the handleDashBoard function when the button is clicked
      >
        Read More
      </Button>
    </div>
  );
};

export default Landing; // Export the Landing component for use in the application
