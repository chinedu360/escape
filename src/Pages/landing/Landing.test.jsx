import { render, screen } from "@testing-library/react";
import Landing from "./Landing";
import { MemoryRouter } from "react-router-dom"; // Use MemoryRouter for routing-related tests
import { vi, describe, it, expect } from "vitest";
import TestRenderer from "react-test-renderer";

describe("testing the Landing/Home Page", () => {
  it("renders landing component", () => {
    render(
      <MemoryRouter>
        <Landing />
      </MemoryRouter>
    );

    //Building assumption varaibles
    const titleElement = screen.getByText("Read An Article Today!");
    const buttonElement = screen.getByText("Read More");

    //running expectation
    expect(titleElement).toBe(titleElement);
    expect(buttonElement).toBeInTheDocument();
  });

  it("test the children inside the Landing component", () => {
    const landing = TestRenderer.create(
      <MemoryRouter>
        <Landing />
      </MemoryRouter>
    ).toJSON();

    //landing containers
    expect(landing.children[0].type).toBe("div");
    expect(landing.children[1].type).toBe("button");
  });

  it("Snapshot", () => {
    const landing = TestRenderer.create(
      <MemoryRouter>
        <Landing />
      </MemoryRouter>
    ).toJSON();

    expect(landing).toMatchSnapshot();
  });
});

// test("navigates to dashboard on button click", () => {
//   const mock = vi.fn();
//   mock.mockImplementation("react-router-dom", () => ({
//     ...vi.importActual("react-router-dom"),
//     useNavigate: () => mock,
//   }));

//   render(
//     <MemoryRouter>
//       <Landing />
//     </MemoryRouter>
//   );

//   const buttonElement = screen.getByText("Read More");
//   expect(buttonElement).toBeInTheDocument();

//   buttonElement.click();

//   // Make assertions on mock
//   // Make assertions on mock
//   // expect(mock.calls?.length).toBe(1); // Check the number of calls
//   expect(mock?.calls[0][0]).toBe("/dashboard"); // Check the first call arguments
// });
