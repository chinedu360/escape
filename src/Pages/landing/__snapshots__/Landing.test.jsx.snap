// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing the Landing/Home Page > Snapshot 1`] = `
<div
  className="flex flex-col justify-center items-center"
>
  <div
    className=""
  >
    <h1
      className="font-bold text-[30px] sm:text-[60px] md:text-[80px]"
    >
      Read An Article Today!
    </h1>
  </div>
  <button
    className="chakra-button css-1uerm0v"
    onClick={[Function]}
    type="button"
  >
    Read More
    <span
      className="chakra-button__icon css-1hzyiq5"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 8a.5.5 0 0 1 .5-.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5A.5.5 0 0 1 4 8z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;
